src/lib/config/root-config.ts

```diff
@@ -88,21 +88,41 @@ export class Endpoint {
   }
 
   static getURLFromConfig = (config: Config): string | undefined => {
-    const url = config.strOpt("url");
-    const scheme = config.strOpt("scheme");
-    const domain = config.strOpt("domain");
-    const port = config.numberOpt("port");
+    return this.getURLInner({
+      url: config.strOpt("url"),
+      scheme: config.strOpt("scheme"),
+      domain: config.strOpt("domain"),
+      port: config.numberOpt("port"),
+    });
+  };
+
+  static getURLFromFlags = (flags: Config): string | undefined => {
+    return this.getURLInner({
+      url: flags.strOpt("endpointURL"),
+      scheme: flags.strOpt("scheme"),
+      domain: flags.strOpt("domain"),
+      port: flags.numberOpt("port"),
+    });
+  };
 
+  static getURLInner = (opts: {
```

Comment:
nit: maybe just `getURL`? the Inner part makes me think there is some like nested url or something when I first read it.

Reply:
makes sense, i'll rename it